pkg/bloomcompactor/sharding.go

```diff
@@ -14,7 +14,7 @@ var (
 // ShardingStrategy describes whether compactor "owns" given user or job.
 type ShardingStrategy interface {
 	util_ring.TenantSharding
-	OwnsJob(job Job) (bool, error)
+	OwnsFp(tenantID string, fp uint64) (bool, error)
```

Comment:
this would map easier to the already existing method in the interface definition we are overriding: https://github.com/grafana/loki/pull/11295/files#diff-6ef3b2d19fc3ebaf9ba3ae94986583eb6d468b74c6b872438ec8832791aca6b5R42